src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -577,7 +582,7 @@ private void RecursivelyApply(Action<ProjectFactory> action, dynamic alreadyAppl
 
                 string fullPath = item.GetMetadataValue("FullPath");
                 if (!string.IsNullOrEmpty(fullPath) &&
-                    !NuspecFileExists(fullPath) &&
+                    !NuspecFileExists(fullPath) && !File.Exists(Path.Combine(Path.GetDirectoryName(fullPath), "project.json")) &&
```

Comment:
This should support projectName.project.json also, take a look at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/ProjectJsonPathUtilities.cs#L28